cmd/notification.go

```diff
@@ -171,6 +173,174 @@ func (sys *NotificationSys) LoadCredentials() map[xnet.Host]error {
 	return errors
 }
 
+// StartProfiling - start profiling on remote peers, by initiating a remote RPC.
+func (sys *NotificationSys) StartProfiling(profiler string) map[xnet.Host]error {
+	errors := make(map[xnet.Host]error)
+	var wg sync.WaitGroup
+	for addr, client := range sys.peerRPCClientMap {
+		wg.Add(1)
+		go func(addr xnet.Host, client *PeerRPCClient) {
+			defer wg.Done()
+			// Try to load credentials in three attempts.
+			for i := 0; i < 3; i++ {
+				err := client.StartProfiling(profiler)
+				if err == nil {
+					break
```

Comment:
The retry here has a problem here, if we encounter an error in the first try, but err == nil in the second try, the error will still be there in maps

Reply:
Ah, good catch.